MSBLOC.Core/Model/ProjectDetails.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace MSBLOC.Core.Services
+{
+    public class ProjectDetails
+    {
+        public string ProjectFile { get; }
+        public string CloneRoot { get; }
+        public string ProjectDirectory { get; }
+        public IReadOnlyDictionary<string, string> Paths => _paths;
+
+        private Dictionary<string, string> _paths;
+
+        public ProjectDetails(string cloneRoot, string projectFile)
+        {
+            CloneRoot = cloneRoot;
+            ProjectFile = projectFile;
+
+            ProjectDirectory = Path.GetDirectoryName(projectFile)
+                               ?? throw new InvalidOperationException(
+                                   "Path.GetDirectoryName(startedEventArgs.ProjectFile) is null");
+
+            _paths = new Dictionary<string, string>();
+        }
+
+        public void AddItems(params string[] itemProjectPaths)
+        {
+            _paths = itemProjectPaths
+                .ToDictionary(item => item, GetClonePath);
+        }
+
+        private string GetClonePath(string itemProjectPath)
+        {
+            return Path.Combine(ProjectDirectory, itemProjectPath)
+                .Split(new[] {CloneRoot}, StringSplitOptions.RemoveEmptyEntries)
+                .First();
+        }
+
+        public string GetPath(string itemProjectPath)
+        {
+            return _paths[itemProjectPath];
```

Comment:
Calling `GetPath` with an `itemProjectPath` that doesn't exist will throw a `KeyNotFoundException` which doesn't make a lot of sense to the user. Should probably test if key exists and throw a custom exception.

Reply:
👍